app/process/handler/crypto-transfer/crypto-transfer_test.go

```diff
@@ -0,0 +1,192 @@
+
+	return ctm, topicID, accID, cryptoTransferPayload, topicSubmissionMessageBytes
+}
+
+func Test_Handle_Not_Initial_Transaction(t *testing.T) {
+	ctm, topicID, accID, cryptoTransferPayload, topicSubmissionMessageBytes := GetTestData()
+	ctHandler, transactionRepo, hederaNodeClient, hederaMirrorClient := InitializeHandler()
+
+	proto.Unmarshal(cryptoTransferPayload, &ctm)
+
+	expectedTransaction := hedera.TransactionID{
+		AccountID:  accID,
+		ValidStart: time.Time{},
+	}
+
+	tx := &transaction.Transaction{
+		Model:          gorm.Model{},
+		TransactionId:  ctm.TransactionId,
+		EthAddress:     ctm.EthAddress,
+		Amount:         ctm.Amount,
+		Fee:            ctm.Fee,
+		Signature:      signature,
+		SubmissionTxId: submissionTxID,
+		Status:         txRepo.StatusCompleted,
+	}
+
+	txs := txn.HederaTransactions{
+		Transactions: []txn.HederaTransaction{},
+	}
+
+	transactionRepo.On("GetByTransactionId", ctm.TransactionId).Return(tx, nil)
+	transactionRepo.On("UpdateStatusSignatureSubmitted", ctm.TransactionId, submissionTxID, signature).Return(nil)
+	hederaNodeClient.On("SubmitTopicConsensusMessage", topicID, topicSubmissionMessageBytes).Return(&expectedTransaction, nil)
+	hederaMirrorClient.On("GetAccountTransaction", submissionTxID).Return(&txs, nil)
+
+	ctHandler.Handle(cryptoTransferPayload)
+
+	transactionRepo.AssertNotCalled(t, "UpdateStatusSignatureSubmitted", ctm.TransactionId, submissionTxID, signature)
+}
+
+func Test_Handle_Initial_Transaction(t *testing.T) {
+	ctm, topicID, accID, cryptoTransferPayload, topicSubmissionMessageBytes := GetTestData()
+	ctHandler, transactionRepo, hederaNodeClient, hederaMirrorClient := InitializeHandler()
+
+	proto.Unmarshal(cryptoTransferPayload, &ctm)
+
+	expectedTransaction := hedera.TransactionID{
+		AccountID:  accID,
+		ValidStart: time.Time{},
+	}
+
+	tx := &transaction.Transaction{
+		Model:          gorm.Model{},
+		TransactionId:  ctm.TransactionId,
+		EthAddress:     ctm.EthAddress,
+		Amount:         ctm.Amount,
+		Fee:            ctm.Fee,
+		Signature:      signature,
+		SubmissionTxId: submissionTxID,
+		Status:         txRepo.StatusInitial,
+	}
+
+	txs := txn.HederaTransactions{
+		Transactions: []txn.HederaTransaction{},
+	}
+
+	transactionRepo.On("GetByTransactionId", ctm.TransactionId).Return(tx, nil)
+	transactionRepo.On("UpdateStatusSignatureSubmitted", ctm.TransactionId, submissionTxID, signature).Return(nil)
+	hederaNodeClient.On("SubmitTopicConsensusMessage", topicID, topicSubmissionMessageBytes).Return(&expectedTransaction, nil)
+	hederaMirrorClient.On("GetAccountTransaction", submissionTxID).Return(&txs, nil)
+
+	ctHandler.Handle(cryptoTransferPayload)
+
+	transactionRepo.AssertCalled(t, "UpdateStatusSignatureSubmitted", ctm.TransactionId, submissionTxID, signature)
```

Comment:
I think that we should assert the following as-well:
- `SubmitTopicConsensusMessage` is called 
- `GetAccountTransaction` is called

This would verify that `handleTopicSubmission` and `checkForTransactionCompletion` were called